### DISCUSSION TITLE
Needs Numeric Field on Product Customization Field

### DISCUSSION THREAD

**Discussion by tuhingr** [PrestaShop/PrestaShop#40152] [Ideas]
I checked there are only text and file can create only product page, 
WE also need numeric field, 
Some digital product has need Mobile Phone IMEI (15 digit IMEI). 
if have numeric field so customer cannot put numeric with text. 
Looking for numeric field look like Phone "IMEI" field 15 digit, and last digit auto created. i:e: https://www.drupal.org/project/imei_field

**Comment by khouloudbelguith** [comment]
Hi @tuhingr,

thanks for submitting your idea!
Just to be sure I understood your need: currently on PrestaShop, customers can personalize the product by entering some text or by providing custom image files.
In addition to this fields, you need to add a new field like a reference for this customization?
Let me know what you think!
Have a nice day,

**Comment by tuhingr** [comment]
I have phone business, some digital product needs phone 15 digit IMEI for Unlock service,
So customer must put numeric code (IMEI No), if field is numeric so customer cannot put any text code.
Thats why need numeric field.

**Comment by khouloudbelguith** [comment]
@tuhingr, thanks for this clarifications.
@marionf, @colinegin what do you think?
Thanks!

**Comment by colinegin** [comment]
Hello,

Thanks for your feedback ! I've seen you have posted your idea on the PrestaShop uservoice, it's a great idea as it will help us prioritize the next features for the following versions.

In the meantime, in the customization part of the product page, you can choose the type Text and in the label field specify that it should only be numeric. It's not perfect, but it might help !

Thank you very much and have a nice day,

Coline

**Comment by marionf** [comment]
We can add an option "Numeric" in the dropdown below

![capture d'écran_1927](https://user-images.githubusercontent.com/13449658/63690911-1e012680-c80e-11e9-8ba6-e381f7aa72e4.png)

**Comment by antoniopineapple** [comment]
Hi there is it possible link the number of text field to the quantity? so for example 1 item > 1 text field
2 item 2 text field appears etc etc
